include/aos/common/tools/error.hpp

```diff
@@ -279,24 +279,48 @@ struct RetWithError {
      * @param value return value.
      * @param error return error.
      */
-    RetWithError(T value, const Error& error)
+    RetWithError(const T& value, const Error& error)
```

Comment:
And again: it should be two separate commits: make RetWithError to handle UniquePtr and add Tie